wordpress-mobile/WordPress-Android
# Feature/15429 domains dashboard logic improvements

Fixes #15429 

This PR does the following 

- Shows Primary domain at the top and below it one of the following
- Shows **Your site domains** List if the site already has custom domain/s
- Shows **Get your domain** if it is a free plan and/or no domain credit.  
- Shows **Claim your free domain** if site has domain credit

This doesn't match with what is listed here entirely p1633549036242800-slack-C0280085R7W.   I'm not sure what to follow here.  Expect to fix it if any in the next PR after clarification.

<table><tr><td><img width=320 src="https://user-images.githubusercontent.com/990349/136909187-a520973c-a59b-4f1d-a169-2dff4fda42e1.png"</td><td><img width=320 src="https://user-images.githubusercontent.com/990349/136332736-0e6d4170-edb0-4496-b43d-24cda05a9dfa.png"</td><td><img width=320 src="https://user-images.githubusercontent.com/990349/136332759-c27c775b-eb17-40e2-b59a-37869ef8def2.png"</td></tr></table>

Fixes #15234 

To test:

- Enable AppSettings -> Debug settings -> SiteDomainsFeatureConfig ✅
- Return to My Site, Notice there's a new item under Configuration -> 🌐 Domains
- Click on Domains, takes you to **Domains** dashboard screen
- Test with a site that has free plan, and a paid plan

To test:

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:
